dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/utils/SdkReference.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.okhttp.utils
+
+import com.datadog.android.Datadog
+import com.datadog.android.v2.api.SdkCore
+import java.util.concurrent.atomic.AtomicReference
+
+internal class SdkReference(
+    private val sdkInstanceName: String? = null,
+    private val onSdkInstanceCaptured: (SdkCore) -> Unit = {}
+) {
+
+    private val reference = AtomicReference<SdkCore>(null)
+
+    fun get(): SdkCore? {
+        val current = reference.get()
+        // TODO RUMM-3195 Check if instance is stopped and remove it?
+        @Suppress("IfThenToElvis") // less readable
+        return if (current == null) {
```

Comment:
I think you still need a `synchronized` block here. That `AtomicReference` will not guarantee you that `Datadog.getInstance(sdkInstanceName)` will not be called multiple times from different threads. That means you will return 2 references even though in the `AtomicReference` you only keep one.
I would add:
```
if(current == null){
 synchronized(this){
     current = reference.get()
     if(current == null) {
                val sdkCore = Datadog.getInstance(sdkInstanceName)
                if (reference.compareAndSet(null, sdkCore)) {
                    onSdkInstanceCaptured(sdkCore)
                }
                sdkCore
     }
 }
```
}

Reply:
We don't need to synchronize here, because `Datadog.initialize` / `Datadog.getInstance` are already synchronized internally. This means that each time the same instance will be returned.

I'm using `compareAndSet` just to make sure that `onSdkInstanceCaptured` will be called once in case of multiple threads.

Reply:
Good to know, tks for the explanation ;)